woocommerce/woocommerce-ios
# REST API: Analytics for Jetpack setup flow

Closes: #8919 


## Description

This PR adds analytics for the Jetpack setup flow with application passwords.
Also added some refactoring:
- Updated JetpackBenefitView and view model to reuse the Jetpack check from the setup coordinator.
- Updated JetpackSetupViewModel to track both cases when the setup is handled during and after login.
- Disabled swipe to dismiss on the wpcom login flow during Jetpack setup.

## Testing instructions

Prerequisite: make sure that you have a self-hosted Woo store without Jetpack.
- On the dashboard screen, tap the Jetpack benefit banner at the bottom or navigate to Menu > Settings > Install Jetpack.
- Tap the button Log in to continue. Notice in Xcode console: `🔵 Tracked jetpack_benfits_login_button_tapped, properties: [AnyHashable("is_wpcom_store"): false, AnyHashable("blog_id"): -1]`.
- When the Jetpack check completes successfully, you should see in Xcode console: `🔵 Tracked jetpack_setup_connection_check_completed, properties: [AnyHashable("blog_id"): -1, AnyHashable("is_wpcom_store"): false, AnyHashable("is_already_connected"): false, AnyHashable("requires_connection_only"): true]` with `is_already_connected` and `requires_connection_only` reflects the correct connection state of your store.
- When the login screen is displayed, a new event is tracked: `🔵 Tracked jetpack_setup_login_flow, properties: [AnyHashable("step"): "email_address", AnyHashable("is_wpcom_store"): false, AnyHashable("blog_id"): -1]`.
- Proceed to log in with incorrect username/password/magic link/2FA code, you should see the correct step, error and submit events being tracked properly following the events listed in #8919.
- When the login completes, the Jetpack setup flow is presented. Then the correct steps should be tracked: `🔵 Tracked jetpack_setup_flow, properties: [AnyHashable("blog_id"): -1, AnyHashable("is_wpcom_store"): false, AnyHashable("step"): "connection"]`. Notice that the events are tracked correctly; no events with the prefix `login_jetpack` are tracked.
- When Jetpack setup completes, you should see event: `🔵 Tracked jetpack_setup_flow, properties: [AnyHashable("blog_id"): -1, AnyHashable("is_wpcom_store"): false, AnyHashable("step"): "all_done"]`.
- Tap the "Go to Store" button, you should then see events with blog ID other than -1, and after all synchronization is done, Xcode console should log: `🔵 Tracked jetpack_setup_synchronization_completed, properties: [AnyHashable("blog_id"): 216745888, AnyHashable("is_wpcom_store"): false]`.

## Screenshots

N/A

---
